### PR TITLE
Feat(server): Add Google Sheets Integration

### PR DESCRIPTION
**Pull request by Torantulino**
This change adds a Google Sheets Integration using a OAuth2 Google Sign-in flow.|

This is a primitive implementation for now, with the authorisation token living purely on the block itself and not being saved elsewhere in the backend, but it works!

- [x] Reader MVP
- [x] Writer MVP
- [x] Read range
- [x] Read single row
- [x] Append single row
- [x] OAuth2 Sign-in with Google
- [x] Read from URL rather than Spreadsheet ID
- [ ] Add tests

### Nice to haves / Future:
- [ ] Persistent Sign-in Cookie?
- [ ] Click sign-in once to be signed into all blocks
- [ ] Overwrite single row?
- [ ] Overwrite range?

image

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        service = build("sheets", "v4", credentials=credentials)
+        sheet = service.spreadsheets()
+        result = sheet.values().get(spreadsheetId=input_data.spreadsheet_id, range=input_data.range).execute()
+        values = result.get("values", [])
+        
+        if not values:
+            yield "data", {"error": "No data found."}
+        else:
+            data = [row for row in values[0:]]
+            yield "rows", data
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py]: You could yield one row at a time similar to the csv one

File: rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        )
+
+    def run(self, input_data: Input) -> BlockOutput:
+        credentials = Credentials(input_data.access_token)
+        service = build("sheets", "v4", credentials=credentials)
+        sheet = service.spreadsheets()
+        result = sheet.values().get(spreadsheetId=input_data.spreadsheet_id, range=input_data.range).execute()
+        values = result.get("values", [])
+        
+        if not values:
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py]: Probably yield error, message

**Comment by Torantulino** [comment]
I can't for the life of me get test mocking to work on this.

File: rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py (hunk 1 above)
Comment (Torantulino) [reply rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py]: What CSV one?

Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py]: I had opened a pr for a csv reader

Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/google-sheets-block.py]: https://github.com/Significant-Gravitas/AutoGPT/pull/7499

**Comment by Pwuts** [comment]
Will fix up this PR and merge it once we have finished implementing initial OAuth support.

**Pull request closed without merging**